[Feature] [Postgre CDC] Unsupported type: ARRAY<INT>

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

使用Zeta引擎，使用Postgre CDC创建连接报错不支持类型，看官方稳定是支持对应的类型的。


### SeaTunnel Version

2.3.4

### SeaTunnel Config

```conf
env {
  execution.parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 300
  read_limit.bytes_per_second=50000000
  read_limit.rows_per_second=10000
}
source {
 Postgres-CDC {
    #result_table_name = "app_task_info"
    username = "sjxx"
    password = "Sjxxxx"
    database-names = ["ccm_xx"]
    schema-names = ["public"]
    table-names = ["ccm_xx.public.app_task_info"]
    base-url = "jdbc:postgresql://xx.xx.xxx.xx:5436/ccm_xx?loggerLevel=OFF"
  }
}

sink {
    Console {
    }
}
```


### Running Command

```shell
../bin/seatunnel.sh -c pg_source2.template
```


### Error Exception

```log
Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:199)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.api.table.factory.FactoryException: ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a source for identifier 'Postgres-CDC'.
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:100)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parseSource(MultipleTableJobConfigParser.java:320)
        at org.apache.seatunnel.engine.core.parse.MultipleTableJobConfigParser.parse(MultipleTableJobConfigParser.java:181)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.getLogicalDag(ClientJobExecutionEnvironment.java:88)
        at org.apache.seatunnel.engine.client.job.ClientJobExecutionEnvironment.execute(ClientJobExecutionEnvironment.java:161)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:146)
        ... 2 more
Caused by: java.lang.UnsupportedOperationException: Unsupported type: ARRAY<INT>
        at org.apache.seatunnel.connectors.cdc.debezium.row.SeaTunnelRowDebeziumDeserializationConverters.createNotNullConverter(SeaTunnelRowDebeziumDeserializationConverters.java:178)
        at org.apache.seatunnel.connectors.cdc.debezium.row.SeaTunnelRowDebeziumDeserializationConverters.createConverter(SeaTunnelRowDebeziumDeserializationConverters.java:113)
        at org.apache.seatunnel.connectors.cdc.debezium.row.SeaTunnelRowDebeziumDeserializationConverters.lambda$new$0(SeaTunnelRowDebeziumDeserializationConverters.java:71)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:546)
        at java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
        at java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:505)
        at org.apache.seatunnel.connectors.cdc.debezium.row.SeaTunnelRowDebeziumDeserializationConverters.<init>(SeaTunnelRowDebeziumDeserializationConverters.java:73)
        at org.apache.seatunnel.connectors.cdc.debezium.row.SeaTunnelRowDebeziumDeserializeSchema.createTableRowConverters(SeaTunnelRowDebeziumDeserializeSchema.java:295)
        at org.apache.seatunnel.connectors.cdc.debezium.row.SeaTunnelRowDebeziumDeserializeSchema.<init>(SeaTunnelRowDebeziumDeserializeSchema.java:87)
        at org.apache.seatunnel.connectors.cdc.debezium.row.SeaTunnelRowDebeziumDeserializeSchema$Builder.build(SeaTunnelRowDebeziumDeserializeSchema.java:331)
        at org.apache.seatunnel.connectors.seatunnel.cdc.postgres.source.PostgresIncrementalSource.createDebeziumDeserializationSchema(PostgresIncrementalSource.java:112)
        at org.apache.seatunnel.connectors.cdc.base.source.IncrementalSource.<init>(IncrementalSource.java:114)
        at org.apache.seatunnel.connectors.seatunnel.cdc.postgres.source.PostgresIncrementalSource.<init>(PostgresIncrementalSource.java:69)
        at org.apache.seatunnel.connectors.seatunnel.cdc.postgres.source.PostgresIncrementalSourceFactory.lambda$createSource$1(PostgresIncrementalSourceFactory.java:103)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:112)
        at org.apache.seatunnel.api.table.factory.FactoryUtil.createAndPrepareSource(FactoryUtil.java:73)
        ... 7 more
```


### Zeta or Flink or Spark Version

Zeta

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
+++ b/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
@@ -17,6 +17,8 @@
 
 package org.apache.seatunnel.connectors.cdc.debezium.row;
 
+import org.apache.seatunnel.shade.com.google.common.annotations.VisibleForTesting;
+
 import org.apache.seatunnel.api.table.type.ArrayType;
 import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
 import org.apache.seatunnel.api.table.type.SeaTunnelRow;
@@ -32,7 +34,6 @@ import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.Struct;
 import org.apache.kafka.connect.source.SourceRecord;
 
-import org.apache.seatunnel.shade.com.google.common.annotations.VisibleForTesting;
 import io.debezium.data.SpecialValueDecimal;
 import io.debezium.data.VariableScaleDecimal;
 import io.debezium.time.MicroTime;

--- a/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
+++ b/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
@@ -17,6 +17,7 @@
 
 package org.apache.seatunnel.connectors.cdc.debezium.row;
 
+import org.apache.seatunnel.api.table.type.ArrayType;
 import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
 import org.apache.seatunnel.api.table.type.SeaTunnelRow;
 import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
@@ -31,6 +32,7 @@ import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.Struct;
 import org.apache.kafka.connect.source.SourceRecord;
 
+import com.google.common.annotations.VisibleForTesting;
 import io.debezium.data.SpecialValueDecimal;
 import io.debezium.data.VariableScaleDecimal;
 import io.debezium.time.MicroTime;
@@ -48,6 +50,7 @@ import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.ZoneId;
 import java.util.Arrays;
+import java.util.List;
 import java.util.Optional;
 
 /** Deserialization schema from Debezium object to {@link SeaTunnelRow} */
@@ -173,10 +176,45 @@ public class SeaTunnelRowDebeziumDeserializationConverters implements Serializab
                 return createRowConverter(
                         (SeaTunnelRowType) type, serverTimeZone, userDefinedConverterFactory);
             case ARRAY:
+                return createArrayConverter(type);
             case MAP:
             default:
                 throw new UnsupportedOperationException("Unsupported type: " + type);
         }
+    }
+
+    @VisibleForTesting
+    protected static DebeziumDeserializationConverter createArrayConverter(
+            SeaTunnelDataType<?> type) {
+        SeaTunnelDataType elementType = ((ArrayType) type).getElementType();
+        switch (elementType.getSqlType()) {
+            case BOOLEAN:
+                return (dbzObj, schema) ->
+                        convertListToArray((List<Boolean>) dbzObj, Boolean.class);
+            case SMALLINT:
+                return (dbzObj, schema) -> convertListToArray((List<Short>) dbzObj, Short.class);
+            case INT:
+                return (dbzObj, schema) ->
+                        convertListToArray((List<Integer>) dbzObj, Integer.class);
+            case BIGINT:
+                return (dbzObj, schema) -> convertListToArray((List<Long>) dbzObj, Long.class);
+            case DOUBLE:
+                return (dbzObj, schema) -> convertListToArray((List<Double>) dbzObj, Double.class);
+            case STRING:
+                return (dbzObj, schema) -> convertListToArray((List<String>) dbzObj, String.class);
+            default:
+                throw new IllegalArgumentException(
+                        "Unsupported SQL type: " + elementType.getSqlType());
+        }
+    }
+
+    private static <T> T[] convertListToArray(List<T> list, Class<T> clazz) {
+        @SuppressWarnings("unchecked")
+        T[] array = (T[]) java.lang.reflect.Array.newInstance(clazz, list.size());
+        for (int i = 0; i < list.size(); i++) {
+            array[i] = list.get(i);
+        }
+        return array;
     }
 
     private static DebeziumDeserializationConverter convertToBoolean() {

--- a/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
+++ b/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
@@ -32,7 +32,7 @@ import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.Struct;
 import org.apache.kafka.connect.source.SourceRecord;
 
-import com.google.common.annotations.VisibleForTesting;
+import org.apache.seatunnel.shade.com.google.common.annotations.VisibleForTesting;
 import io.debezium.data.SpecialValueDecimal;
 import io.debezium.data.VariableScaleDecimal;
 import io.debezium.time.MicroTime;
@@ -208,8 +208,8 @@ public class SeaTunnelRowDebeziumDeserializationConverters implements Serializab
         }
     }
 
+    @SuppressWarnings("unchecked")
     private static <T> T[] convertListToArray(List<T> list, Class<T> clazz) {
-        @SuppressWarnings("unchecked")
         T[] array = (T[]) java.lang.reflect.Array.newInstance(clazz, list.size());
         for (int i = 0; i < list.size(); i++) {
             array[i] = list.get(i);

--- a/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
+++ b/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConverters.java
@@ -199,6 +199,8 @@ public class SeaTunnelRowDebeziumDeserializationConverters implements Serializab
                         convertListToArray((List<Integer>) dbzObj, Integer.class);
             case BIGINT:
                 return (dbzObj, schema) -> convertListToArray((List<Long>) dbzObj, Long.class);
+            case FLOAT:
+                return (dbzObj, schema) -> convertListToArray((List<Float>) dbzObj, Float.class);
             case DOUBLE:
                 return (dbzObj, schema) -> convertListToArray((List<Double>) dbzObj, Double.class);
             case STRING:

--- a/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/test/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConvertersTest.java
+++ b/seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/test/java/org/apache/seatunnel/connectors/cdc/debezium/row/SeaTunnelRowDebeziumDeserializationConvertersTest.java
@@ -17,10 +17,12 @@
 
 package org.apache.seatunnel.connectors.cdc.debezium.row;
 
+import org.apache.seatunnel.api.table.type.ArrayType;
 import org.apache.seatunnel.api.table.type.BasicType;
 import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
 import org.apache.seatunnel.api.table.type.SeaTunnelRow;
 import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.connectors.cdc.debezium.DebeziumDeserializationConverter;
 import org.apache.seatunnel.connectors.cdc.debezium.DebeziumDeserializationConverterFactory;
 import org.apache.seatunnel.connectors.cdc.debezium.MetadataConverter;
 
@@ -34,6 +36,7 @@ import org.junit.jupiter.api.Test;
 
 import java.time.ZoneId;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
 
 public class SeaTunnelRowDebeziumDeserializationConvertersTest {
@@ -74,5 +77,54 @@ public class SeaTunnelRowDebeziumDeserializationConvertersTest {
         SeaTunnelRow row = converters.convert(record, value, schema);
         Assertions.assertEquals(row.getField(0), 1);
         Assertions.assertNull(row.getField(1));
+    }
+
+    @Test
+    void testArrayConverter() throws Exception {
+        DebeziumDeserializationConverter converter;
+        // bool array converter
+        converter =
+                SeaTunnelRowDebeziumDeserializationConverters.createArrayConverter(
+                        ArrayType.BOOLEAN_ARRAY_TYPE);
+        Boolean[] booleans = new Boolean[] {false, true};
+        Assertions.assertTrue(
+                Arrays.equals(
+                        booleans, (Boolean[]) (converter.convert(Arrays.asList(booleans), null))));
+        // smallInt array converter
+        converter =
+                SeaTunnelRowDebeziumDeserializationConverters.createArrayConverter(
+                        ArrayType.SHORT_ARRAY_TYPE);
+        Short[] shorts = new Short[] {(short) 1, (short) 2};
+        Assertions.assertTrue(
+                Arrays.equals(shorts, (Short[]) (converter.convert(Arrays.asList(shorts), null))));
+        // int array converter
+        converter =
+                SeaTunnelRowDebeziumDeserializationConverters.createArrayConverter(
+                        ArrayType.INT_ARRAY_TYPE);
+        Integer[] ints = new Integer[] {1, 2};
+        Assertions.assertTrue(
+                Arrays.equals(ints, (Integer[]) (converter.convert(Arrays.asList(ints), null))));
+        // long array converter
+        converter =
+                SeaTunnelRowDebeziumDeserializationConverters.createArrayConverter(
+                        ArrayType.LONG_ARRAY_TYPE);
+        Long[] longs = new Long[] {1L, 2L};
+        Assertions.assertTrue(
+                Arrays.equals(longs, (Long[]) (converter.convert(Arrays.asList(longs), null))));
+        // float array converter
+        converter =
+                SeaTunnelRowDebeziumDeserializationConverters.createArrayConverter(
+                        ArrayType.FLOAT_ARRAY_TYPE);
+        Float[] floats = new Float[] {1.0f, 2.0f};
+        Assertions.assertTrue(
+                Arrays.equals(floats, (Float[]) (converter.convert(Arrays.asList(floats), null))));
+        // double array converter
+        converter =
+                SeaTunnelRowDebeziumDeserializationConverters.createArrayConverter(
+                        ArrayType.DOUBLE_ARRAY_TYPE);
+        Double[] doubles = new Double[] {1.0, 2.0};
+        Assertions.assertTrue(
+                Arrays.equals(
+                        doubles, (Double[]) (converter.convert(Arrays.asList(doubles), null))));
     }
 }

